tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs

```diff
@@ -141,6 +143,7 @@ public async Task ConfidentialClientWithCertificateTestAsync()
             MsalAssert.AssertAuthResult(authResult);
             appCacheRecorder.AssertAccessCounts(2, 1);
             Assert.IsTrue(appCacheRecorder.LastNotificationArgs.IsApplicationCache);
+            Assert.AreEqual(AppCacheKey, appCacheRecorder.LastNotificationArgs.SuggestedCacheKey);
```

Comment:
Integration tests seem to only cover `AcquireTokenForClient` and `OBO`, but not `AcquireTokenSilent`. 
We need some unit test coverage as well - should be easy to update some existing tests for Client Creds and OBO. For AcquireTokenSilent as well. For `GetAccount` I think you need a dedicated unit test.